Vector/ViewController/SettingsViewController.m

```diff
@@ -2180,6 +2222,71 @@ - (void)onProfileAvatarTap:(UITapGestureRecognizer *)recognizer
     [self presentViewController:navigationController animated:YES completion:nil];
 }
 
+- (void)exportEncryptionKeys:(UITapGestureRecognizer *)recognizer
+{
+    [currentAlert dismiss:NO];
+
+    MXKEncryptionKeysExportView *exportView = [[MXKEncryptionKeysExportView alloc] initWithMatrixSession:self.mainSession];
+    currentAlert = exportView;
+
+    // Use a temporary file for the export
+    keyExportsFile = [NSURL fileURLWithPath:[NSTemporaryDirectory() stringByAppendingPathComponent:@"riot-keys.txt"]];
+
+    // Make sure the file is empty
+    [self deleteKeyExportFile];
+
+    // Show the export dialog
+    __weak typeof(self) weakSelf = self;
+    [exportView showInViewController:self toExportKeysToFile:keyExportsFile onComplete:^(BOOL success) {
+
+        if (weakSelf && success)
+        {
+             typeof(self) self = weakSelf;
+            self->currentAlert = nil;
+
+            // Let another app handling this file
+            self->documentInteractionController = [UIDocumentInteractionController interactionControllerWithURL:keyExportsFile];
+            [self->documentInteractionController setDelegate:self];
+
+            if ([self->documentInteractionController presentOptionsMenuFromRect:self.view.frame inView:self.view animated:YES])
+            {
+                // We want to delete the temp keys file after it has been processed by the other app.
+                // We use [UIDocumentInteractionControllerDelegate didEndSendingToApplication] for that
+                // but it is not reliable for all cases (see http://stackoverflow.com/a/21867096).
+                // So, arm a timer to auto delete the file after 10mins.
+                keyExportsFileDeletionTimer = [[NSTimer alloc] initWithFireDate:[NSDate dateWithTimeIntervalSinceNow:(10 * 60)]
```

Comment:
Why don't you use [NSTimer scheduledTimerWithTimeInterval:...]?
this creates and returns a new NSTimer object and schedules it on the current run loop in the default mode.